### ISSUE TITLE
On Firefox presence_of_all_elements_located does not raise UnexpectedAlertPresentException

### ISSUE THREAD

**Issue by avladevic** [SeleniumHQ/selenium#6891]
## 🐛 Bug Report

On Firefox waiting for `presence_of_all_elements_located` does not rise `UnexpectedAlertPresentException` when alert is shown.

Also it does not matter if alert is shown before or after page load.
Does work on Chrome as expected.

## To Reproduce

Load a page which shows an alert then wait for some element.

## Expected behavior

`UnexpectedAlertPresentException` should be raised.

## Test script or set of commands reproducing this issue
**_before-page-load.html_**
```before-page-load.html
<html>
    <head>
        <script>alert('Hello!');</script>
    </head>
    <body>
        <div id="test">Test</div>
    </body>
</html>
```

**_after-page-load.html_**
```after-page-load.html
<html>
    <head>
        <script>
	    window.onload = function(){
	        setTimeout(function(){alert('Hello!');}, 0);
	    }
        </script>
    </head>
    <body>
        <div id="test">Test</div>
    </body>
</html>
```

**_test_unexpected_alert.py_**
```test_unexpected_alert.py
from unittest import TestCase

from selenium import webdriver
from selenium.common.exceptions import UnexpectedAlertPresentException
from selenium.webdriver.common.by import By
from selenium.webdriver.support.expected_conditions import presence_of_all_elements_located, presence_of_element_located
from selenium.webdriver.support.wait import WebDriverWait

class UnexpectedAlertTestCase(TestCase):
    def setUp(self):
        self.driver = webdriver.Firefox()

    def tearDown(self):
        if hasattr(self, 'driver') and self.driver:
            self.driver.quit()

    def test_presence_all_alert_before_page_load(self):
        # This test fails because it raises TimeoutException instead of UnexpectedAlertPresentException!
        self.driver.get('https://laughing-beaver-1bd1b5.netlify.com/before-page-load.html')

        wait = WebDriverWait(self.driver, 2)

        with self.assertRaises(UnexpectedAlertPresentException):
            wait.until(presence_of_all_elements_located((By.ID, 'test')))

    def test_presence_all_alert_after_page_load(self):
        # This test fails because wait returns the element instead of rising UnexpectedAlertPresentException!
        self.driver.get('https://laughing-beaver-1bd1b5.netlify.com/after-page-load.html')

        wait = WebDriverWait(self.driver, 2)

        with self.assertRaises(UnexpectedAlertPresentException):
            # Here the wait method returns the WebElement
            wait.until(presence_of_all_elements_located((By.ID, 'test')))

    def test_presence_single_before_page_load(self):
        # This test is fine!
        self.driver.get('https://laughing-beaver-1bd1b5.netlify.com/before-page-load.html')

        wait = WebDriverWait(self.driver, 2)

        with self.assertRaises(UnexpectedAlertPresentException):
            wait.until(presence_of_element_located((By.ID, 'test')))

    def test_presence_single_after_page_load(self):
        # This test is fine!
        self.driver.get('https://laughing-beaver-1bd1b5.netlify.com/after-page-load.html')

        wait = WebDriverWait(self.driver, 2)

        with self.assertRaises(UnexpectedAlertPresentException):
            wait.until(presence_of_element_located((By.ID, 'test')))
```

## Environment

OS: Ubuntu 18.04.1
Browser: Firefox
Browser version: 64.0.2
Browser Driver version: GeckoDriver 0.23
Language Bindings version: Python 3.141

**Comment by lmtierney** [comment]
Hi I was just getting around to trying this. I'm getting the first one failing still but not the last 3. Have you raised this issue with the geckodriver team?

**Comment by avladevic** [comment]
Hi I test it again. This issue is fixed in latest version of Firefox (67.0.3). Still failing on 64.0.2. I think we can close the issue.
